sympy/controlsys/lti.py

```diff
@@ -0,0 +1,1363 @@
+The second call will return a list of matrix valued outputs, matching the time input.
+
+For details on syntax and additional methods, see the docstings of each class and its methods.
+
+"""
+from __future__ import division
+
+from sympy import (
+    Symbol, var, simplify, oo, exp,
+    Poly, lcm, LC, degree, Integral, integrate,
+    Matrix, BlockMatrix, eye, zeros,
+    latex, ShapeError, ImmutableMatrix, MutableMatrix,
+    SparseMatrix, MutableDenseMatrix
+)
+
+from sympy.core.compatibility import range
+from sympy.printing import sstr
+
+# import mpmath for numercial results
+from mpmath import expm, quad, matrix as mpm_matrix
+
+__all__ = [
+    'StateSpaceModel',
+    'TransferFunctionModel'
+]
+
+_matrixTypes = (
+    Matrix, ImmutableMatrix, MutableMatrix, SparseMatrix, MutableDenseMatrix)
+
+
+class StateSpaceModel(object):
+    """state space model (ssm) of a linear, time invariant control system
+
+    Represents the standard state-space model with state matrix A, input matrix B, output matrix C, and
+    transmission matrix D. This makes the linear control system:
+        (1) x'(t) = A * x(t) + B * u(t);    x in R^n , u in R^k
+        (2) y(t)  = C * x(t) + D * u(t);    y in R^m
+    where u(t)  is any input signal, y(t) the corresponding output, and x(t) the systems state.
+
+    Parameters
+    ==========
+
+    *arg : TransferFunctionModel, Sympy-Matrices
+        tfm to construct the state space model from, or the Matrices A,B,C,D as StateSpaceModel(A, B, C, D)
+
+    Examples
+    ========
+
+    >>> from sympy import Matrix, Symbol
+    >>> from sympy.controlsys.lti import StateSpaceModel, TransferFunctionModel
+
+    The easiest way to create a StateSpaceModel is via four matrices:
+
+    >>> A, B, C, D = Matrix([1,2]), Matrix([2,3]), Matrix([2]), Matrix([0])
+    >>> StateSpaceModel(A, B, C, D)
+    StateSpaceModel(
+    Matrix([
+    [1],
+    [2]]),
+    Matrix([
+    [2],
+    [3]]),
+    Matrix([[2]]),
+    Matrix([[0]]))
+
+    One can use less matrices as well. The rest will be filled with a minimum of
+    zeros:
+
+    >>> StateSpaceModel(A, B)
+    StateSpaceModel(
+    Matrix([
+    [1],
+    [2]]),
+    Matrix([
+    [2],
+    [3]]),
+    Matrix([[0]]),
+    Matrix([[0]]))
+
+    One can also use a TansferFunctionModel to create a StateSpaceModel
+
+    >>> s = Symbol('s')
+    >>> tfm = TransferFunctionModel(Matrix([1/s, s/(1 + s**2)]))
+    >>> StateSpaceModel(tfm)
+    StateSpaceModel(
+    Matrix([
+    [0, -1, 0],
+    [1,  0, 0],
+    [0,  1, 0]]),
+    Matrix([
+    [1],
+    [0],
+    [0]]),
+    Matrix([
+    [1, 0, 1],
+    [1, 0, 0]]),
+    Matrix([
+    [0],
+    [0]]))
+
+    See Also
+    ========
+
+    TransferFunctionModel: transfer function model of a lti system
+
+    References
+    ==========
+
+    Joao P. Hespanha, Linear Systems Theory. 2009.
+    """
+
+    def __init__(self, *arg):
+
+        def zero():
+            self.represent[0] = zeros(1)
```

Comment:
The `represent` attribute is not explained, doesn't have an obvious meaning for the name.